manim/mobject/polyhedra.py

```diff
@@ -0,0 +1,361 @@
+    """
+
+    def __init__(
+        self,
+        vertex_coords: List[np.ndarray],
+        faces_list: List[List[Hashable]],
+        faces_config: dict = {},
+        graph_config: dict = {},
+    ):
+        VGroup.__init__(self)
+        self.faces_config = dict(
+            {"fill_opacity": 0.5, "shade_in_3d": True}, **faces_config
+        )
+        self.graph_config = dict(
+            {
+                "vertex_type": Dot3D,
+                "edge_config": {
+                    "stroke_opacity": 0,  # I find that having the edges visible makes the polyhedra look weird
+                },
+            },
+            **graph_config
+        )
+        self.vertex_coords = vertex_coords
+        self.vertex_indices = list(range(len(self.vertex_coords)))
+        self.layout = dict(enumerate(self.vertex_coords))
+        self.faces_list = faces_list
+        self.face_coords = [[self.layout[j] for j in i] for i in faces_list]
+        self.edges = self.get_edges(self.faces_list)
+        self.faces = self.create_faces(self.face_coords)
+        self.graph = Graph(
+            self.vertex_indices, self.edges, layout=self.layout, **self.graph_config
+        )
+        self.add(self.faces, self.graph)
+        self.add_updater(self.update_faces)
+
+    def get_edges(self, faces_list):
+        """Creates list of cyclic pairwise tuples."""
+        edges = []
+        for face in faces_list:
+            edges += zip(face, face[1:] + face[:1])
+        return edges
+
+    def create_faces(self, face_coords):
+        """Creates VGroup of faces from a list of face coordinates."""
+        face_group = VGroup()
+        for face in face_coords:
+            face_group.add(Polygon(*face, **self.faces_config))
+        return face_group
+
+    def update_faces(self, m):
+        face_coords = self.extract_face_coords()
+        new_faces = self.create_faces(face_coords)
+        self.faces.match_points(new_faces)
+
+    def extract_face_coords(self):
+        """Extracts the coordinates of the vertices in the graph.
+        Used for updating faces.
+        """
+        new_vertex_coords = []
+        for v in self.graph.vertices:
+            new_vertex_coords.append(self.graph[v].get_center())
+        layout = dict(enumerate(new_vertex_coords))
+        return [[layout[j] for j in i] for i in self.faces_list]
+
+
+class Tetrahedron(Polyhedron):
+    """A tetrahedron, one of the five platonic solids. It has 4 faces, 6 edges, and 4 vertices.
+
+    Parameters
+    ----------
+    edge_length:
+        The length of an edge between any two vertices.
+
+    Examples
+    --------
+
+    .. manim:: TetrahedronScene
+        :save_last_frame:
+
+        class TetrahedronScene(ThreeDScene):
+            def construct(self):
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                obj = Tetrahedron()
+                self.add(obj)
+    """
+
+    def __init__(self, edge_length: float = 1, **kwargs):
```

Comment:
Shouldn't `edge_length` be either `float` or `int`?
```suggestion
    def __init__(self, edge_length: typing.Union[float, int] = 1, **kwargs):
```

Reply:
As there is the import statement `from typing import Hashable, List, Tuple, Union`
it should be `Union` instead of `typing.Union`

Reply:
changed